### ISSUE TITLE
Changing the output file names is missing in English

### ISSUE THREAD

**Issue by gfnool** [withastro/docs#10348]
### 📚 Subject area/topic

Configuring Astro

### 📋 Suggested page

https://docs.astro.build/en/guides/configuring-astro/

### 📋 General description or bullet points (if proposing new content)

In the Italian version there is a section about "Changing the output file names"
https://docs.astro.build/it/guides/configuring-astro/#cambiare-il-nome-dei-file-di-output
That section is missing in the English version, and I think it's very useful.

**Comment by delucis** [comment]
This was recently removed (the Italian needs updating to remove it too).

As these are standard Rollup options ([`assetFileNames`](https://rollupjs.org/configuration-options/#output-assetfilenames), [`chunkFileNames`](https://rollupjs.org/configuration-options/#output-chunkfilenames), and [`entryFileNames`](https://rollupjs.org/configuration-options/#output-entryfilenames)), not unique to Astro, and it’s a fairly advanced use case (most users should not need to configure this), we decided to remove it from this page which is focused on introducing basic concepts around configuring Astro.

If we get feedback that this content was valuable though, we could consider restoring the content in some other place (like a dedicated recipe). In that case, it would be helpful to hear what use cases people have for doing this.

**Comment by sarah11918** [comment]
Yes, as Chris said, we determined that the information was definitely not in the correct spot on the new introductory page, and we have a task to consider whether this information is indeed something for Astro to document.

I do think this could make a recipe, in which case we would need someone to construct a proper structure around it. Things like:

- state a problem to solve/specific goal to accomplish (e.g. change output filenames specifically to these names, no "general theory")
- write an opinionated step-by-step example that achieves that goal
- show how to check that implementing it was successful

The original English entry, with the code example, can be found [in the old v4 docs](https://v4.docs.astro.build/en/guides/configuring-astro/#customising-output-filenames). If someone would like to turn this into a recipe, we would happily take a PR!

**Comment by gfnool** [comment]
OK, I understand your points.

However, I still think it would be useful to explain why filenames like this appear:
`Layout.astro_astro_type_script_index_0_lang.DulELbMk.js`
even for something as simple as `<script src="../scripts/index.js"></script>`.

I’ve noticed that in other Astro builds I’ve seen, this doesn’t happen. As someone new to Astro, it feels unusual and a bit confusing, and I’d like to understand the reasoning behind this behavior and how to resolve it to avoid such strange and lengthy filenames.

Thank you for your kind reply

**Comment by delucis** [comment]
> it would be useful to explain why filenames like this appear:
> `Layout.astro_astro_type_script_index_0_lang.DulELbMk.js`
> even for something as simple as `<script src="../scripts/index.js"></script>`

Is that when running `astro build`? That looks like a dev server file name to me.

**Comment by delucis** [comment]
Ah, yes, looks like this is not so uncommon. For example, the docs site currently loads `/_astro/Search.astro_astro_type_script_index_0_lang.ByHJUbA9.js`. I misremembered those names being a bit shorter.

The short story is: Astro takes care of bundling and optimizing your `<script>` tags and these optimised files get some automatic names based on a few factors (I’m not super familiar with all the reasons).

But on the whole it shouldn’t matter too much: everything in the `_astro/` directory is managed by Astro, linked to by Astro etc. One advantage for example is the “hash” in the names means these files can be cached by web browsers indefinitely. If ever the script changes, it would have a new hash in the name, forcing a redownload. That’s good for site performance.

**Comment by sarah11918** [comment]
Pinging for freshness! This content mostly exists from past versions of the docs and we would welcome someone writing a recipe that is a pretty simple "How to do this" and doesn't need much explanation!
